### PR TITLE
vaapic: memset misc object

### PR DESCRIPTION
**Pull request by lizhong1008**
Signed-off-by: Zhong Li <zhong.li@intel.com>

### CODE REVIEW

**Review by chivakker — CHANGES_REQUESTED** [review]
The change makes sense but the commit message is rather vague.  There's no such a thing as vaapic, vaapi-encpicture has to be cleared to 0 when creating a miscellaneous object.

File: encoder/vaapiencpicture.h [hunk 1]
```diff
@@ -124,6 +124,8 @@ createMiscObject(VAEncMiscParameterType miscType, T * &bufPtr)
     if (obj) {
         misc->type = miscType;
         bufPtr = (T *) (misc->data);
+        if (bufPtr)
```
Comment (xuguangxin) [inline encoder/vaapiencpicture.h]: This if is not needed, data is an array in misc, if we can access misc->type, misc->data never be NULL.

Comment (lizhong1008) [reply encoder/vaapiencpicture.h]: @xuguangxin misc is an input of createBufferObject,  and drive will init misc->data, may change it to NULL.  So we should check bufPtr

Comment (xuguangxin) [reply encoder/vaapiencpicture.h]: The data's prototype is array.  misc->data will get the array address. if misc is not null, misc->data always be a valid value.

**Comment by lizhong1008** [comment]
@chivakker  yeah, your comments make sense, it is a typo. I'll update it.

**Review by chivakker — APPROVED** [review]

**Comment by wudping** [comment]
verify + 1

**Comment by xuguangxin** [comment]
merged as 0ee515f10124740a8f0d9830210be5d8d2142387

**Pull request closed without merging**